src/Solnet.Rpc/Core/Http/JsonRpcClient.cs

```diff
@@ -157,5 +157,106 @@ private async Task<RequestResult<T>> HandleResult<T>(JsonRpcRequest req, HttpRes
 
             return result;
         }
+
+        /// <summary>
+        /// Sends a batch of messages as a POST method and returns a collection of responses.
+        /// </summary>
+        /// <param name="reqs">The message request.</param>
+        /// <returns>A task that represents the asynchronous operation that holds the request result.</returns>
+        public async Task<RequestResult<JsonRpcBatchResponse>> SendBatchRequest(JsonRpcBatchRequest reqs)
+        {
+            var requestsJson = JsonSerializer.Serialize(reqs, _serializerOptions);
+
+            try
+            {
+                // TODO - logging
+
+                // create byte buffer to avoid charset=utf-8 in content-type header
+                // as this is rejected by some RPC nodes
+                var buffer = Encoding.UTF8.GetBytes(requestsJson);
+                using var httpReq = new HttpRequestMessage(HttpMethod.Post, "/")
+                {
+                    Content = new ByteArrayContent(buffer)
+                    {
+                        Headers = {
+                            { "Content-Type", "application/json"}
+                        }
+                    }
+                };
+
+                // execute POST
+                using (var response = await _httpClient.SendAsync(httpReq).ConfigureAwait(false))
+                {
+                    var result = await HandleBatchResult(reqs, response).ConfigureAwait(false);
+                    result.RawRpcRequest = requestsJson;
+                    return result;
+                }
+
+            }
+            catch (HttpRequestException e)
+            {
+                var result = new RequestResult<JsonRpcBatchResponse>(e.StatusCode ?? System.Net.HttpStatusCode.BadRequest, e.Message);
+                result.RawRpcRequest = requestsJson;
+                // TODO - logging
+                return result;
+            }
+            catch (Exception e)
+            {
+                var result = new RequestResult<JsonRpcBatchResponse>(System.Net.HttpStatusCode.BadRequest, e.Message);
+                result.RawRpcRequest = requestsJson;
+                // TODO - logging

```

Comment:
Same as base Solnet
```chsarp
_logger?.LogDebug(new EventId(req.Id, req.Method), $"Caught exception: {e.Message}");
```

Reply:
Please could you give me a steer on the best thing to use to construct the `EventId` with?
The batch contains multiple ids (for each inner request) but there isn't an overall id used for the batch itself.

Would it be ok to construct a composite id that includes the range? e.g. 4...23 - this may be inperfect as there could be gaps but is perhaps less crap than just choosing the first id.

Reply:
Gone with minimum request id as the id and `[batch on n]` as the event method.
Let me know if you'd like this changing...